inspirehep/modules/orcid/tasks.py

```diff
@@ -185,40 +189,61 @@ def attempt_push(orcid, rec_id, oauth_token):
         rec_id(int): inspire record's id to push to ORCID.
         oauth_token(string): orcid token.
     """
-    put_code = get_putcode_from_redis(orcid, rec_id)
+    put_code, previous_hash = get_putcode_and_hash_from_redis(orcid, rec_id)
 
     if not put_code:
-        record_put_codes = get_author_putcodes(orcid, oauth_token)
+        recache_all_author_putcodes(orcid, oauth_token)
+        put_code, previous_hash = get_putcode_and_hash_from_redis(orcid, rec_id)
 
-        for fetched_rec_id, fetched_put_code in record_put_codes:
-            store_record_in_redis(orcid, fetched_rec_id, fetched_put_code)
+    new_code, new_hash = push_record_with_orcid(
+        recid=str(rec_id),
+        orcid=orcid,
+        oauth_token=oauth_token,
+        put_code=put_code,
+        old_hash=previous_hash,
+    )
 
-        put_code = get_putcode_from_redis(orcid, rec_id)
+    if new_code != put_code or new_hash != previous_hash:
+        store_record_in_redis(orcid, rec_id, new_code, new_hash)
 
-    new_code = push_record_with_orcid(str(rec_id), orcid, oauth_token, put_code)
 
-    if not put_code:
-        store_record_in_redis(orcid, rec_id, new_code)
+def recache_all_author_putcodes(orcid, oauth_token):
+    """Fetch all putcodes from ORCID and cache them.
+
+    Args:
+        orcid(string): an orcid identifier.
+        oauth_token(string): orcid token.
+    """
+    record_put_codes = get_author_putcodes(orcid, oauth_token)
+
+    for fetched_rec_id, fetched_put_code in record_put_codes:
+        store_record_in_redis(orcid, fetched_rec_id, fetched_put_code, None)
 
 
-def store_record_in_redis(orcid, rec_id, put_code):
+def store_record_in_redis(orcid, rec_id, put_code, hashed):
```

Comment:
`hashed=None`, so you don't have to pass an explicit `None` when you're not storing it.